### PR TITLE
Add scale transformation support to score_model_out()

### LINKED ISSUE #48: handle scale transformations for scoring
mild complication: which output types support which transformations?

split this out from #13 out of laziness/not wanting to think this through.

### LINKED ISSUE #91: Add log transformation for scoring
Generally, we're interested in adding a way to calculate scores (like WIS) on a log scale. This might involve writing a new helper function that could be called in `score_model_out()` to transform the forecasts on a log scale before scoring them. Note that we can maybe make use of [`scoringutils::transform_forecasts()`](https://epiforecasts.io/scoringutils/reference/transform_forecasts.html). Other issues to reference include #13 and (this one's parent) #48

### PR DESCRIPTION
**Pull request by annakrystalli**
## Summary

Adds `transform`, `transform_append`, and `transform_label` arguments to `score_model_out()` for computing scores on transformed scales (e.g., log, sqrt).

Closes #48, closes #91

## Design decisions

- **Supported output types**: Only quantile, mean, and median. PMF forecasts error with an informative message since scale transformations don't make sense for categorical data.

- **Implementation**: Passes through to `scoringutils::transform_forecasts()` rather than implementing custom transformation logic.

- **Label inference**: The `transform_label` is auto-generated from the function name using `rlang::enexpr()` (e.g., `sqrt` → "sqrt", `scoringutils::log_shift` → "log_shift"). Anonymous functions require an explicit label.

- **When labels matter**: The label only appears in output when `transform_append=TRUE`, where it distinguishes transformed rows from "natural" rows in the `scale` column. We still require labels for anonymous functions for consistency, but document when it's actually used.

## New parameters

- `transform`: A transformation function (e.g., `sqrt`, `scoringutils::log_shift`)
- `transform_append`: If `TRUE`, compute scores on both original and transformed scales
- `transform_label`: Optional custom label (auto-generated if not provided)

### CODE REVIEW

**Comment by annakrystalli** [comment]
Note: Running lintr on this PR shows some pre-existing issues unrelated to these changes. Opened #93 to track fixing them separately.

File: R/score_model_out.R [hunk 1]
```diff
@@ -144,6 +169,18 @@ score_model_out <- function(
...
 
+  # Apply scale transformation if requested
+  if (!is.null(transform)) {
+    # Use the captured expression for label inference (e.g., "log_shift")
+    label <- get_transform_label(transform_expr, transform_label)
+    su_data <- scoringutils::transform_forecasts(
+      su_data,
+      fun = transform,
+      append = transform_append,
+      label = label
```
Comment (nickreich) [inline R/score_model_out.R]: I think we want to include a `...` argument here so an offset, or other transformation, parameter could be passed to the `transform_forecasts()` function

File: R/score_model_out.R [hunk 2]
```diff
@@ -28,6 +28,24 @@
...
+#'   If `NULL` (the default), no transformation is applied. Only supported for
+#'   quantile, mean, and median output types.
+#' @param transform_append Logical. If `FALSE` (the default), scores are
+#'   computed only on the transformed scale. If `TRUE`, scores are computed on
+#'   both original and transformed scales, with a `scale` column distinguishing
+#'   them.
+#' @param transform_label A character string label for the transformation
+#'   (e.g., "log"). If `NULL` (the default), the label is auto-generated from
+#'   the function name (e.g., "log_shift" for `scoringutils::log_shift`).
+#'   Required when using an anonymous transform function. Note: the label only
```
Comment (nickreich) [inline R/score_model_out.R]: what is an "anonymous transform function"?

Comment (nickreich) [reply R/score_model_out.R]: I lean towards making this required if `transform_append = TRUE` and then avoiding the complexities of "inferring" a label. They require it in scoringutils with no default or inference. Let's do the same here and keep it simple.

File: tests/testthat/test-score_model_out.R [hunk 3]
```diff
@@ -614,3 +614,144 @@ test_that("score_model_out errors when an unsupported output_type is provided",
     regexp = "only supports the following types"
   )
 })
+
+
+# Tests for scale transformation functionality
+
+test_that("transform argument produces same scores as manually transformed data", {
```
Comment (nickreich) [inline tests/testthat/test-score_model_out.R]: I think it would also be good to test this same logic out for log_shift with an offset.

File: tests/testthat/test-score_model_out.R [hunk 4]
```diff
@@ -614,3 +614,144 @@ test_that("score_model_out errors when an unsupported output_type is provided",
...
+    by = "model_id"
+  )
+
+  # Both approaches should produce identical scores
+
+  expect_equal(scores_manual$wis, scores_auto$wis)
+})
+
+
+test_that("score_model_out succeeds with log transformation on median forecasts", {
```
Comment (nickreich) [inline tests/testthat/test-score_model_out.R]: could we add a test case where there is a zero forecast and log-transform is used without an offset. I think we should expect a warning there (or we should follow scoringutils lead on how to handle that setting).

File: R/score_model_out.R (hunk 1 above)
Comment (annakrystalli) [reply R/score_model_out.R]: Good catch! Done in 50ff306. Added `...` to `score_model_out()` which passes through to `transform_forecasts()`, enabling use of `offset` and `base` arguments with `log_shift` (and any other transform function parameters).

Also added tests for:
- Verifying `...` passthrough works (using `offset = 1` avoids NaN scores on data with zeros)
- Confirming warnings from scoringutils are propagated (e.g., "Detected zeros in input values")
